translator/translate/otel/processor/jmxprocessor/translator.go

```diff
@@ -0,0 +1,156 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package jmxprocessor
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/filterprocessor"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/confmap"
+	"go.opentelemetry.io/collector/processor"
+
+	"github.com/aws/amazon-cloudwatch-agent/translator/translate/otel/common"
+)
+
+const (
+	regexp = "regexp"
+)
+
+var (
+	jmxKey     = common.ConfigKey(common.MetricsKey, common.MetricsCollectedKey, common.JmxKey)
+	configKeys = map[component.DataType]string{
+		component.DataTypeMetrics: common.ConfigKey(jmxKey),
+	}
+
+	jmxTargets = []string{"activemq", "cassandra", "hbase", "hadoop", "jetty", "jvm", "kafka", "kafka-consumer", "kafka-producer", "solr", "tomcat", "wildfly"}
+)
+
+type translator struct {
+	dataType component.DataType
+	name     string
+	index    int
+	factory  processor.Factory
+}
+
+type Option interface {
+	apply(t *translator)
+}
+
+type optionFunc func(t *translator)
+
+func (o optionFunc) apply(t *translator) {
+	o(t)
+}
+
+// WithDataType determines where the translator should look to find
+// the configuration.
+func WithDataType(dataType component.DataType) Option {
+	return optionFunc(func(t *translator) {
+		t.dataType = dataType
+	})
+}
+
+func WithIndex(index int) Option {
+	return optionFunc(func(t *translator) {
+		t.index = index
+	})
+}
+
+var _ common.Translator[component.Config] = (*translator)(nil)
+
+func NewTranslator(opts ...Option) common.Translator[component.Config] {
+	return NewTranslatorWithName("", opts...)
+}
+
+func NewTranslatorWithName(name string, opts ...Option) common.Translator[component.Config] {
+	t := &translator{name: name, index: -1, factory: filterprocessor.NewFactory()}
+	for _, opt := range opts {
+		opt.apply(t)
+	}
+	if name == "" && t.dataType != "" {
```

Comment:
Does not need a `dataType`. Set the name to `jmx` instead, so it's `filter/jmx/0`.